### DISCUSSION TITLE
Active active DR with MM2

### DISCUSSION THREAD

**Discussion by Gulnaz2021** [strimzi/strimzi-kafka-operator#4600] [Q&A]
Hi,

We are trying to achieve active active DR with MM2, and below are the steps and observations:
Steps:
1) Deployed kafka cluster1 in source ns1
2) Deployed kafka cluster2 in source ns2
3) Deployed MM2 in ns1
4) Deployed MM2 in ns2
5) Created a "topic" in ns1, it gets replicated to ns2 by MM2 as "cluster1.topic"
6) Produced messages to "topic" in ns1, consumed messages from "cluster1.topic" in ns2 - successfully do that
7) Produced messages to "cluster1.topic" in ns2, trying to consume messages from "topic" in ns1 - but here from ns1 topic we are not able to consume the messages, that were produced to the replicated topic in ns2, 
and what we are able to consume from topic in ns1, are the messages that were produced to it in step 6)

We need a capability to produce and consume message from the second cluster despite of the topic being created in the first. The idea is say first cluster goes down, we should still be able to produce and consume from the secondary cluster, and after that if secondary cluster goes down, we should be able to produce and consume from the primary cluster.

Please let us know, how can we achieve the active active DR, with the above scenario.
Thanks.

**Comment by scholzj** [comment]
You should probably read the MM2 docs first. Lot of it is explained for example here: https://strimzi.io/docs/operators/latest/full/using.html#bidirectional_replication_activeactive ... I'm sure even more stuff will be in Kafka docs for example. You should be aslo able to find recordings of conference talks describing how MM2 works.

**Comment by Gulnaz2021** [comment]
Hi Jacob,

Went through the document you've shared, but as mentioned for the step 7,
Produced messages to "cluster1.topic" in ns2, trying to consume messages from "topic" in ns1 - but here from ns1 topic we are not able to consume the messages, that were produced to the replicated topic in ns2,
and what we are able to consume from topic in ns1, are the messages that were produced to it in step 6)

we are trying to achieve the DR with active active kafka clusters.

In the document, it says there are 2 topics present with same name in cluster1 and cluster2, and they are replicated accordingly to the clusters using MM2. So, please find below the questions that I have:
1) Does it mean that, only one topic created in the cluster1, is replicated to the cluster2, and again to cluster1, or 
2) Separate topics have been created with same name in cluster1 and cluster2, and have been replicated to the adjacent clusters using MM2.

**Reply by scholzj** [reply]
As shown in the picture in the documentation - there is always only one topic which is writable in each cluster. So the producer always produces to the `topic` which is mirrored to the other cluster. Never to the already mirrored topic.
